openthread/CMakeLists.txt

```diff
@@ -6,6 +6,19 @@
 include(${NRFXLIB_DIR}/common.cmake)
 nrfxlib_calculate_lib_path(lib_path)
 
+file (READ "../../modules/lib/openthread/include/openthread/instance.h" INSTANCE_FILE_CONTENT)
```

Comment:
This whole check needs to be placed underneath `if (CONFIG_OPENTHREAD_LIBRARY_1_1)`, otherwise the warning will be printed even when the library is not used. Plus I'd consider making a cmake function out of this code (in case it would be needed to verify API version for 1.2 library).